hw2/04-background.js

```diff
@@ -1 +1,41 @@
-// Add your code here
+const interval = document.getElementById('interval');
+const button = document.getElementById('toggle');
+let buttonState = false;
+
+addEventListener('load', () => {
+  // loading random background color on page load
+  randomColor();
+});
+
+// creating a setInterval id per MDN docs
+let id;
+
+button.addEventListener('click', (event) => {
+  event.preventDefault(); // not sure if this necessary? todo: ask prof
+
+  // on click change buttonState
+  buttonState = !buttonState;
+  button.classList.toggle('btn-primary', !buttonState);
+  button.classList.toggle('btn-danger', buttonState);
+  button.value = buttonState ? 'Stop' : 'Start';
+
+  /* if nothing set, set interval to 3 seconds, and
+       multiply value in input by 1000 (for milliseconds) */
+  if (buttonState) {
+    let intervalAmount = interval.value == '' ? 3000 : interval.value * 1000;
```

Comment:
Similar to my feedback in exercise 3 you might want to change the == to ===. I also agree with your use of a ternary. the airbnb guide says to keep those to one line, simple and not nested. I think you have good use of it here.